app/controllers/history.js

```diff
@@ -3,8 +3,12 @@ import Ember from 'ember';
 export default Ember.Controller.extend({
 
   sortProperties: ['name:desc'],
+  queryParams: ['q'],
+  q: '',
   sortedModel: Ember.computed.sort('model', 'sortProperties'),
-
+  filteredModel: Ember.computed.filter('sortedModel', function(item) {
+    return item.name.toLowerCase().includes(this.get('q'));
```

Comment:
If I use any capitalization in the search term, it won't find anything. You might want to  convert `q` to lowercase as well.